content/blogs/2024-01-08-sentry-plugin.md

```diff
@@ -0,0 +1,550 @@
+    payload: |
+      {
+          "message": "Kestra Monthly Data Purge FAIL",
+          "level": "error"
+      }
+
+triggers:
+  - id: monthly
+    type: io.kestra.core.models.triggers.types.Schedule
+    cron: "0 9 1 * *" # every month at 9am on the 1st day of the month
+```
+
+### SentryExecution Task
+`type: io.kestra.plugin.notifications.sentry.SentryExecution`
+
+This task demonstrates a really cool feature of Kestra: triggers that trigger other triggers. Let me explain.
+
+
+If you have just one flow with some non-zero likelihood of failure - you add a SentryAlert like the examples above. But what do you do when you have 10 or 20 flows? You definitely do not want to have to maintain 20 individual SentryAlert tasks. **That’s when you use the SentryExecution task because it can triggered by any other flow.**
+
+
+Let’s imagine three flows, **foo**, **bar** and **baz** all related to your company’s payroll; you need to know if any one of these falls over. Each one of these three flows has the same namespace: **payroll**.
+
+```yaml
+id: foo
+namespace: payroll
+
+id: bar
+namespace: payroll
+
+id: baz
+namespace: payroll
+```
+
+Now we create a fourth flow using the SentryExecution task that will be triggered when any flow with the **payroll** namespace fails. This flow has two parts, the task and the trigger.
+
+The task is straightforward. All you need to provide is the **DSN** and the **Level**: 
+
+```yaml
+tasks:
+- id: send_alert
+  type: io.kestra.plugin.notifications.sentry.SentryExecution
+  executionId: "{{ trigger.executionId }}"
+  transaction: "/execution/id/{{ trigger.executionId }}"
+  dsn: "{{ secret('SENTRY_DSN') }}"
+  level: ERROR
+```
+
+The trigger is only slightly more involved: it needs two pieces of information: the condition that will trigger this flow, and the namespace that Kestra will use to link all four flows together. 
+
+```yaml
+triggers:
+  - id: failed_prod_workflows
+    type: io.kestra.core.models.triggers.types.Flow
+    conditions:
+      - type: io.kestra.core.models.conditions.types.ExecutionStatusCondition
+        in:
+          - FAILED
+          - WARNING
+      - type: io.kestra.core.models.conditions.types.ExecutionNamespaceCondition
+        namespace: payroll
+        prefix: false
+```
+
+In the YAML above, the `ExecutionStatusCondition` says that any time the execution status is either `FAILED` or `WARNING`, 
+and the `ExecutionNamespaceCondition` says that if the flow namespace matches payroll exactly, then the `send_alert` task will be executed.  
+
+In pseudo code, the entire flow might read like this:
+
+Execute send_alert task
+  When the flow execution status is equal to FAILED or WARNING
+  And the flow namespace equals payroll exactly.
+
+There’s also a handy variation, using prefix: true. In pseudo code, it would read like this:
+  Execute send_alert task
+  When the flow execution status is equal to FAILED or WARNING
+  And the flow namespace startsWith payroll in lower case
```

Comment:
```suggestion
In the YAML above, the `ExecutionStatusCondition` says that any time the execution status is either `FAILED` or `WARNING`, and the `ExecutionNamespaceCondition` says that if the flow namespace matches payroll exactly, then the `send_alert` task will be executed.  

In pseudo-code, the entire flow might read like this:

Execute `send_alert` task
  When the flow execution status is equal to FAILED or WARNING
  And the flow namespace equals `payroll`.

There’s also a handy variation using the `prefix: true`. In pseudo-code, it would read like this:

  Execute send_alert task
  When the flow execution status is equal to FAILED or WARNING
  And the flow namespace starts with `payroll` in lowercase
```